File: typescript/problems/2733_neither_minimum_nor_maximum.ts
// Given an integer array nums containing distinct positive integers, find and return any number from the array that is neither the minimum nor the maximum value in the array, or -1 if there is no such number.

// Return the selected integer.

function findNonMinOrMax(nums: number[]): number {
    const maxN: number = Math.max(...nums);
    const minN: number = Math.min(...nums);

    for (const n of nums) {
        if (n !== maxN && n !== minN) {
            return n
        }
    }
    return -1
};

File: arrays_and_hashing/problems/2733_neither_minimum_nor_maximum.py
# Given an integer array nums containing distinct positive integers, find and return any number from the array that is neither the minimum nor the maximum value in the array, or -1 if there is no such number.

# Return the selected integer.

class Solution:
    def findNonMinOrMax(self, nums: List[int]) -> int:
        maxN = max(nums)
        minN = min(nums)

        for n in nums:
            if n is not maxN and n is not minN:
                return n
        return -1
        